crypto/evp/p5_crpt.c

```diff
@@ -91,7 +91,7 @@ int PKCS5_PBE_keyivgen(EVP_CIPHER_CTX *cctx, const char *pass, int passlen,
                                             (char *)mdname, 0);
     *p = OSSL_PARAM_construct_end();
     if (EVP_KDF_CTX_set_params(kctx, params) != 1
-            || EVP_KDF_derive(kctx, out, kl) != 1)
```

Comment:
you may be able to pass the params into the derive here..

Reply:
It's possible as of last night.